setup/src/Magento/Setup/Console/CommandLoader.php

```diff
@@ -0,0 +1,110 @@
+<?php
+declare(strict_types=1);
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Setup\Console;
+
+use Laminas\ServiceManager\ServiceManager;
+use Symfony\Component\Console\Command\Command as SymfonyCommand;
+use Symfony\Component\Console\CommandLoader\CommandLoaderInterface;
+use Symfony\Component\Console\Exception\CommandNotFoundException;
+
+/**
+ * Class CommandList contains predefined list of commands for Setup.
+ *
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class CommandLoader implements CommandLoaderInterface
```

Comment:
I believe it can be removed... My goal was to minimize potential BC breaks by leaving the class in place, but since I'm not initializing it in the CLI (https://github.com/magento/magento2/pull/29355/files/2895c6e436fb42480f8f50795066e85af1da2f51#diff-4eb287b4e24c054a400e5f7b40dab0cbL146-L149), it wouldn't be *true* backwards compatibility.

Reply:
Resolved in 2d99944

Reply:
I guess this should be fine, the developers always depend on `CommandListInterface`, also the Setup CommandList doesn't accept any command via dependency injection.